### PR TITLE
75 gdi dataset discovery service retrieve organization details

### PR DESCRIPTION
**Pull request by sulejmank**
## 🚀 Pull Request Checklist

- **Title:**

  - `[ ]` A brief, descriptive title for the changes.

- **Description:**

  - `[ ]` Provide a clear and concise description of your pull request, including the purpose of the changes and the approach you've taken.

- **Context:**

  - `[ ]` Why are these changes necessary? What problem do they solve? Link any related issues.

- **Changes:**

  - `[ ]` List the major changes you've made, ideally organized by commit or feature.

- **Testing:**

  - `[ ]` Describe how the changes have been tested. Include any relevant details about the testing environment and the test cases.

- **Screenshots (if applicable):**

  - `[ ]` If your changes are visual, include screenshots to help explain your changes.

- **Additional Information:**

  - `[ ]` Add any other information that might be useful for reviewers, such as considerations, discussions, or dependencies.

- **Checklist:**
  - `[ ]` I have checked that my code adheres to the project's style guidelines and that my code is well-commented.
  - `[ ]` I have performed self-review of my own code and corrected any misspellings.
  - `[ ]` I have made corresponding changes to the documentation (if applicable).
  - `[ ]` My changes generate no new warnings or errors.
  - `[ ]` I have added tests that prove my fix is effective or that my feature works.
  - `[ ]` New and existing unit tests pass locally with my changes.

## Summary by Sourcery

Introduce functionality to retrieve and include organization details in the dataset discovery service, including new API endpoints, services, repositories, and mappers. Extend dataset schema and update mappers to handle organization data. Add corresponding unit and integration tests.

New Features:
- Introduce an API endpoint to retrieve a list of organizations.
- Add a new service to handle the retrieval of organizations from the CKAN API.
- Implement a repository to interact with the CKAN API for fetching organization data.
- Create a mapper to convert CKAN organization data to the internal dataset organization model.

Enhancements:
- Extend the existing dataset schema to include organization details.
- Update the dataset mappers to include organization information.

Tests:
- Add unit tests for the new organization retrieval service and repository.
- Add integration tests for the organization retrieval API endpoint.

### CODE REVIEW

File: src/main/openapi/discovery.yaml [hunk 1]
```diff
@@ -351,6 +373,20 @@ components:
...
+        name:
+          type: string
+          title: Name of organization
+        title:
+          type: string
+          title: Title of organization
+        description:
+          type: string
+          title: Description of organization
+        image_url:
```
Comment (zalborzi) [inline src/main/openapi/discovery.yaml:387]: this is an url pointing to ckan? if this is the case, we cannot hide ckan behind twingate as we were thinking of

Comment (brunopacheco1) [reply src/main/openapi/discovery.yaml:387]: important: I suggest we bring this discussion to another story, as the network layer must be transparent to the application layer.

Comment (sulejmank) [reply src/main/openapi/discovery.yaml:387]: not sure, but I would say yes. Probably like media link on ckan. Then I guess we cannot have image on organization

**Review by zalborzi — APPROVED** [review]
LGTM

**Comment by brunopacheco1** [comment]
@sulejmank I discovered that `title` is optional as well.

Can you please make empty value as `null`?

`GET /api/v1/organizations` should list objects with `name`, `title`, `description` and `imageUrl`.

**Comment by brunopacheco1** [comment]
@sulejmank as aligned, I created this story to tackle these comments: https://github.com/GenomicDataInfrastructure/gdi-userportal-ckanext-gdi-userportal/issues/85

**Pull request merged**